module-1/src/Components/Counter/Counter.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { render,screen } from "@testing-library/react";
+import user from '@testing-library/user-event';
+import Counter from "./Counter";
+
+describe("Counter Component Tests",()=>{
+
+    test("renders initial value provided in props",()=>{
+        const initialValue = 2; 
+        render(<Counter initialValue={initialValue} />);
+        const valueElement = screen.getByText(`Counter: ${initialValue}`);
+        expect(valueElement).toBeInTheDocument();
+    });
+
+    test("click event on decrement button decrements the displayed value",async ()=>{
+        let initialValue = 1; 
+        user.setup()
+        render(<Counter initialValue={initialValue} />);
+        const decrementButton = screen.getByRole('button',{name: "Decrement"});
+        await user.click(decrementButton).then(
+            ()=>{
+                initialValue--;
+                const valueElement = screen.getByText(`Counter: ${initialValue}`);
+                expect(valueElement).toBeInTheDocument();
+                expect(valueElement).toHaveTextContent("Counter: 0");
+            }
+        );
```

Comment:
Not related to the test logic but you don't need `then` with `await`, that's the entire point of async-await -- instead of chaining promises you write streamlined code `await`ing promises. 
```suggestion
        await user.click(decrementButton);
        initialValue--;
        const valueElement = screen.getByText(`Counter: ${initialValue}`);
        expect(valueElement).toBeInTheDocument();
        expect(valueElement).toHaveTextContent("Counter: 0");
```

Reply:
You're right, I updated the tests and they keeps running fine